### ISSUE TITLE
Requesting positions

### ISSUE THREAD

**Issue by gutschein** [nbartels/vlx2mqtt#4]
Hi Norbert,
I just tried your docker solution: Works great within minutes. Thankx. 

Anyway I've two remarks
* The example-CFG (vlx2mqtt.cfg_demo) uses #-remarks for MQTT login/pw _in the same line_ as the value. That was missleading, as it took some time to figure out that this is not allowed... ;)  Maybe it's better to move the remarks into separate lines
* Would be helpfull to have an additional option to just request to send the current position without SET, but with a topic "GET" or just with the node-name.

Kind regards,
Carsten

**Comment by gutschein** [comment]
Hi Norbert,

there is another reason why it would make sense to have an option "vlx2mqtt/get" forcing to send all positions: If io-homecontrol-RF-senders are used beside MQTT the new position after using the RF is not provided/pushed to MQTT from the KLF200 if there is no option to request frequently all positions without moving the shades. 

As an alternativ the script itself has an option to request all positions frequently from the KLF200 and push them via MQTT. If an additional parameter is 0 or not set no frequent action will be triggered, any other value e.g. in seconds would request frequently.

I appreciate your stable solution. It would be perfect with this tiny enhancement... :)

Kind regards,
Carsten

**Comment by nbartels** [comment]
Sorry, I somehow missed your request ...
I'll check the possibility to trigger a re-send of the last position. I only had a very short look at the PyVLX lib and there seems to be no get_position method, but maybe it is possible to simulate this ... 
Currently, my Openhab saves the last value and waits for new incoming position changes, so I did not run in the problem, but why not improve the script ...

**Comment by gutschein** [comment]
Don't worry, it's just hobby, not business... :)

On my side it's edomi, not Openhub. It saves the last positions, too. But how get Openhab the proper position if the shades were moved via RF ("Handsender")? Sometimes my wife decides that its more convinient to use the RF...
Would be great to have a solution for that cases sometimes. 
Thank you!

**Comment by nbartels** [comment]
Openhab uses two topics for its mqtt items: 
* the command topic to set a new value 
* the state topic to get the current state

In my configuration I use something like this:
```
Type rollershutter : position "Shutter Position" [ 
                stateTopic="vlx2mqtt/Badezimmer/position", 
                commandTopic="vlx2mqtt/Badezimmer/set" ]
``` 

Using the RF, let the KLF send the new position to vlx2mqtt and the script propagates it to mqtt. So as long as you subscribe to the `position` topic you should receive all the changes no matter how they are triggered.

**Comment by gutschein** [comment]
I agree. Sounds logical. I do subscribe to positions and use the same topics like you... :)

But...there seems to be a different behaviour: If I move the shades via RF I do not get a new position from KLF/MQTT. If it would send a new position in this cases, there would be this issue... 
Strange. I've to check this again. Or did you change anything in the docker? If not the behaviour should be the same on your side and my side...

**Comment by gutschein** [comment]
I just tried it again: Just using MQTT-Explorer and my RF. The topic vlx2mqtt/AZ/position is not updated if I use the RF ("Handsender"). The KLF does not push it nor the Docker Script request it.

If I use MQTT to set a position, the position comes back immediatly. If I closed it to e.g. 15% via RF and afterwards setting a position via MQTT, the current position is actulized just in the second of publishing the set-command.

Update: I just realized...I just was to unpatient - just waiting for a while: After 1-2 minutes the position is pushed even if the shade was moved via RF only. So there is no real issue anymore... 😄   
Best issues are self-solving issues...

BTW: Thank your for adjusting the CFG with inline-comment. Maybe it would be helpfull to adjust it in the conatiner description, too. Just to avoid this tiny pitfall.
